tests/unit/lib/hook/test_hook_config.py

```diff
@@ -0,0 +1,105 @@
+"""Test Hook Package Config"""
+import json
+from copy import deepcopy
+from unittest import TestCase
+from unittest.mock import patch, MagicMock, mock_open
+
+from samcli.lib.hook.hook_config import HookFunctionality, HookPackageConfig
+from samcli.lib.hook.exceptions import InvalidHookPackageConfigException
+
+TEST_HOOK_PACKAGE_CONFIG = {
+    "hook_package_id": "my_test_hook_package_id",
+    "hook_use_case": "IaC",
```

Comment:
this example made me think, should we require iac_framework if the use case is IaC?
I think we can use dependentRequired: https://json-schema.org/understanding-json-schema/reference/conditionals.html#id4

Reply:
oh wait I think dependentRequired is only for when the depending property is present but not that it equals some specific value